### PR TITLE
cancellable: use sync.Once to prevent double channel close

### PR DESCRIPTION
**Pull request by LK4D4**
ping @tonistiigi

### CODE REVIEW

**Comment by tonistiigi** [comment]
@LK4D4 Would it make sense to just remove `notifyingReader` if it's not used for anything?

**Comment by LK4D4** [comment]
@tonistiigi I noticed that that `Do` uses it. It waits on that channel there.

File: client/transport/cancellable/cancellable.go [hunk 1]
```diff
@@ -91,23 +92,24 @@ func Do(ctx context.Context, client transport.Sender, req *http.Request) (*http.
...
+		r.notifyOnce.Do(func() {
+			close(r.notify)
+		})
 	}
 	return n, err
 }
 
 func (r *notifyingReader) Close() error {
 	err := r.ReadCloser.Close()
-	if r.notify != nil {
```
Comment (mlaventure) [inline client/transport/cancellable/cancellable.go]: Shouldn't `r.notify` should still be check for `nil`? There's no warranty that the field was assigned to, is there?

Comment (tonistiigi) [inline client/transport/cancellable/cancellable.go]: It's not part of a public api.

Comment (LK4D4) [inline client/transport/cancellable/cancellable.go]: @mlaventure yes, @tonistiigi is right. Even more, it's used only in this file.

**Comment by vdemeester** [comment]
LGTM :cat:

**Pull request merged**